File: 프로그래머스/lv1/86491. 최소직사각형/최소직사각형.py
def solution(sizes):
    min_width = 0
    min_height = 0
    
    for size in sizes:
        width1, width2 = max(min_width, size[0]), max(min_width, size[1])
        height1, height2 = max(min_height, size[1]), max(min_height, size[0])
        
        min_width, min_height = (width1, height1) if width1*height1 < width2*height2 else (width2, height2)
    
    return min_width * min_height

File: 프로그래머스/1/86491. 최소직사각형/최소직사각형.java
class Solution {
    public int solution(int[][] sizes) {
        int width = 0;
        int height = 0;
        
        for (int[] size: sizes) {
            int w = size[0];
            int h = size[1];
            
            if (w < h) {
                int temp = h;
                h = w;
                w = temp;
            }
            
            width = Math.max(width, w);
            height = Math.max(height, h);
        }
        
        return width * height;
    }
}